lib/generateSource.ts

```diff
@@ -35,7 +35,7 @@ export class ${typeName} extends TypeChainContract {
     static async createAndValidate(web3: any, address: string | BigNumber): Promise<${typeName}> {
       const contract = new ${typeName}(web3, address);
       const code = await promisify(web3.eth.getCode, [address]);
-      if (code === "0x0") {
+      if (code.length < 4) {
```

Comment:
I would not accept this PR without a comment on what is the reasoning behind it. The comment should explain all possible cases of this, why anything longer than 3 chars is a correct smart contract and possibly link / reference the issue #50 here on github for people to understand in the future why checking for `0x0` is not sufficient. In an ideal world there would even be a regression test with metamasks `web3` to make sure nobody changes this again to the behaviour before. This was a **very dangerous** bug, because when the check passes on an non-existent contract, one will potentially call some methods anyhow and waste real ether.

Reply:
@Levino I also wanted to leave a comment here but then realized that it's generated code and somehow didn't want to do so. 

It's not possible to have valid smart contract with 3 characters - that's my reasoning for this check. And i think it covers various cases like `0x0` or `0x` or something even else that we are not aware of in a better way.

Reply:
@Levino comment about empty smart contract check was added by https://github.com/Neufund/TypeChain/pull/55. 

Currently, test verifying this behavior is non-existing, we are accepting PRs :)